xline/src/server/command.rs

```diff
@@ -481,6 +481,29 @@ impl ConflictCheck for Command {
             return true;
         }
 
+        if this_req.is_compaction_request() && other_req.is_compaction_request() {
+            return true;
+        }
+
+        if (this_req.is_txn_request() && other_req.is_compaction_request())
+            || (this_req.is_compaction_request() && other_req.is_txn_request())
+        {
+            match (this_req, other_req) {
+                (
+                    &RequestWrapper::CompactionRequest(ref com_req),
+                    &RequestWrapper::TxnRequest(ref txn_req),
+                )
+                | (
+                    &RequestWrapper::TxnRequest(ref txn_req),
+                    &RequestWrapper::CompactionRequest(ref com_req),
+                ) => {
+                    let target_revision = com_req.revision;
+                    return txn_req.is_conflict_with_rev(target_revision)
```

Comment:
Please don't return at here, if `txn_req` is not conflict with revision, we also need to check other conflict after this

Reply:
But this doesn't matter since the conflicts between a `CompactionRequest` and a `TxnRequest` only occur due to revision.